### PR TITLE
Even lazier node instantiation.

### PR DESCRIPTION
**Pull request by jMyles**
A precursor to #1881, this PR integrates the changes in https://github.com/nucypher/bytestringSplitter/pull/27.

### CODE REVIEW

File: nucypher/network/nodes.py [hunk 1]
```diff
@@ -239,17 +237,17 @@ class NodeSprout(PartiallyKwargifiedBytes):
 
     def __init__(self, node_metadata):
         super().__init__(node_metadata)
-        self.checksum_address = to_checksum_address(node_metadata['public_address'][0])
+        self.checksum_address = to_checksum_address(self.public_address)
```
Comment (KPrasch) [inline nucypher/network/nodes.py:240]: :tada: - Looking excellent.

Comment (jMyles) [reply nucypher/network/nodes.py:240]: ...and much more optimal.  Saves a keccak and other smaller cpu-bound operations compared to the previous method.

Comment (tuxxy) [reply nucypher/network/nodes.py:240]: Really nice

File: nucypher/network/nodes.py [hunk 2]
```diff
@@ -461,30 +459,17 @@ def remember_node(self,
...
-        certificate_filepath = self.node_storage.store_node_certificate(certificate=stranger_certificate)
+            # Store node's certificate - It has been seen.
+            certificate_filepath = self.node_storage.store_node_certificate(certificate=stranger_certificate)
 
-        # In some cases (seed nodes or other temp stored certs),
-        # this will update the filepath from the temp location to this one.
-        node.certificate_filepath = certificate_filepath
+            # In some cases (seed nodes or other temp stored certs),
+            # this will update the filepath from the temp location to this one.
+            node.certificate_filepath = certificate_filepath
```
Comment (tuxxy) [inline nucypher/network/nodes.py:471]: This whole thing is looking really nice now.

File: nucypher/network/nodes.py [hunk 3]
```diff
@@ -1235,10 +1225,10 @@ def verify_node(self,
...
         sprout = self.internal_splitter(node_bytes, partial=True)
 
-        verifying_keys_match = sprout['verifying_key'] == self.public_keys(SigningPower)
-        encrypting_keys_match = sprout['encrypting_key'] == self.public_keys(DecryptingPower)
-        addresses_match = sprout['public_address'] == self.canonical_public_address
-        evidence_matches = sprout['decentralized_identity_evidence'] == self.__decentralized_identity_evidence
+        verifying_keys_match = sprout.verifying_key == self.public_keys(SigningPower)
+        encrypting_keys_match = sprout.encrypting_key == self.public_keys(DecryptingPower)
+        addresses_match = sprout.public_address == self.canonical_public_address
+        evidence_matches = sprout.decentralized_identity_evidence == self.__decentralized_identity_evidence
```
Comment (tuxxy) [inline nucypher/network/nodes.py:1231]: :rocket:

File: nucypher/network/nodes.py (hunk 1 above)
Comment (michwill) [reply nucypher/network/nodes.py:240]: With other changes too, readability is certainly improved!

File: tests/config/test_storages.py [hunk 4]
```diff
@@ -63,7 +63,11 @@ def _read_and_write_metadata(self, ursula, node_storage):
         all_stored_nodes = node_storage.all(federated_only=True)
         all_known_nodes.add(ursula)
         assert len(all_known_nodes) == len(all_stored_nodes)
-        assert all_stored_nodes == all_known_nodes
+
+        known_checksums = sorted([n.checksum_address for n in all_known_nodes])
```
Comment (michwill) [inline tests/config/test_storages.py:67]: I think, `[` and `]` are not necessary here

File: tests/learning/test_domains.py [hunk 5]
```diff
@@ -39,4 +39,4 @@ def test_learner_learns_about_domains_separately(ursula_federated_test_config, c
         assert not set(second_domain_learners).intersection(set(new_first_domain_learner.known_nodes))
 
         # However, it learned about *all* of the nodes in its own domain.
-        assert set(first_domain_learners).intersection(set(new_first_domain_learner.known_nodes)) == first_domain_learners
+        assert set(first_domain_learners).intersection(set(n.mature() for n in new_first_domain_learner.known_nodes)) == first_domain_learners
```
Comment (michwill) [inline tests/learning/test_domains.py:42]: Curiously, it works as `.intersection(n.mature() for n in new_first_domain_learner.known_nodes)`, too

File: tests/config/test_storages.py (hunk 4 above)
Comment (jMyles) [reply tests/config/test_storages.py:67]: Indeed; good catch.  I will include this change in #1881.

File: tests/learning/test_domains.py (hunk 5 above)
Comment (jMyles) [reply tests/learning/test_domains.py:42]: Interesting - that's unexpected.  But very nice.  I will also include this in #1881.

**Pull request merged**